source/DataNode.cpp

```diff
@@ -221,6 +221,47 @@ bool DataNode::IsNumber(const string &token)
 
 
 
+// Convert the token at the given index to a boolean. This returns false
+// and prints an error if the index is out of range or the token cannot
+// be interpreted as a number.
+bool DataNode::BoolValue(int index) const
+{
+	// Check for empty strings and out-of-bounds indices.
+	if(static_cast<size_t>(index) >= tokens.size() || tokens[index].empty())
+		PrintTrace("Error: Requested token index (" + to_string(index) + ") is out of bounds:");
+	else if(!IsBool(tokens[index]))
+		PrintTrace("Error: Cannot convert value \"" + tokens[index] + "\" to a boolean:");
+	else
+	{
+		const string &token = tokens[index];
+		return token == "true" || token == "1";
+	}
+
+	return 0.;
+}
+
+
+
+// Check if the token at the given index is a boolean, i.e. "true"/"1" or "false"/"0"
+// as a string.
```

Comment:
This gets me thinking; dont we use never as a bool to say false?

Reply:
If by that you mean that adding something like "invisible" to a mission means "invisible = true," then yeah, but we also didn't have functions like these that made having "true" and "false" tokens make sense. And gamerules is a perfect place to have explicit "false" boolean values, as we want the key to be visible so that people know it exists, instead of being forced to just not list the key in order to have a false value.

Reply:
In missions, maybe. It sounds a little odd here though, like `"universal ramscoop" never`, `"death benefits" never`, `"universal ramscoop" always`, `"death benefits" always`. That kinda make it sounds like there's something in between, could it be "sometimes"? or "50%"? While "true" or "false" is just that, either true or false, nothing else in between. Since this is another mechanic to missions entirely we probably don't necessary have to maintain that consistency. Unless we want to have 50% chance you'll get universal ramscoop today or something.